### PR TITLE
[WIP]Clone query

### PR DESCRIPTION
**Pull request by PolRod**

### CODE REVIEW

File: app/assets/javascripts/angular/directives/query/query_details_directive.js.es6 [hunk 1]
```diff
@@ -3,14 +3,15 @@
 
   class QueryDetailsController {
     constructor(QueryHandler, QueryTab, TagResource, defaultDateFormat, DefaultAceConfigurator, OpenReplService,
-      RoleModel) {
+      RoleModel, $window) {
```
Comment (dollschasingmen) [inline app/assets/javascripts/angular/directives/query/query_details_directive.js.es6]: for navigating, prefer the `$location` service over `$window`

https://github.com/lumoslabs/aleph/blob/master/app/assets/javascripts/angular/services/query/query_handler.js.es6#L29

also we might want to just put this in query_handler service

File: app/assets/javascripts/angular/directives/query/query_details_directive.js.es6 [hunk 2]
```diff
@@ -35,6 +36,18 @@
       this.resultRunner.run().then(this._tabs.navigateToTab.bind(this._tabs, 'results'));
     }
 
+    cloneQuery() {
+      let clone = angular.copy(this.query);
+      clone._item.id = undefined;
```
Comment (dollschasingmen) [inline app/assets/javascripts/angular/directives/query/query_details_directive.js.es6]: lets encapsulate this logic in the `models` lib as a service like we talked about

File: app/assets/javascripts/angular/directives/query/query_details_directive.js.es6 [hunk 3]
```diff
@@ -35,6 +35,18 @@
       this.resultRunner.run().then(this._tabs.navigateToTab.bind(this._tabs, 'results'));
     }
 
+    cloneQuery() {
+      let clone = this.query.clone();
+      clone._item.title = "Copy of " + clone._item.title;
+      let _this = this;
+      clone.save().then(
+        function (serverQueryItem) {
```
Comment (dollschasingmen) [inline app/assets/javascripts/angular/directives/query/query_details_directive.js.es6]: use es6 style arrow syntax anonymous functions

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

File: app/assets/javascripts/angular/directives/query/query_details_directive.js.es6 [hunk 4]
```diff
@@ -35,6 +35,18 @@
       this.resultRunner.run().then(this._tabs.navigateToTab.bind(this._tabs, 'results'));
     }
 
+    cloneQuery() {
+      let clone = this.query.clone();
+      clone._item.title = "Copy of " + clone._item.title;
+      let _this = this;
```
Comment (dollschasingmen) [inline app/assets/javascripts/angular/directives/query/query_details_directive.js.es6]: are u sure you need to do this?

Comment (PolRod) [inline app/assets/javascripts/angular/directives/query/query_details_directive.js.es6]: I think so, if not, this is undefined the next time I want to use it

File: app/assets/javascripts/angular/directives/query/query_details_directive.js.es6 (hunk 3 above)
Comment (PolRod) [inline app/assets/javascripts/angular/directives/query/query_details_directive.js.es6]: Thanks!

**Comment by dollschasingmen** [comment]
its looking pretty good man. put some tests in!

**Comment by dollschasingmen** [comment]
sweet -- let me pull, demo to myself just to make sure the UX makes sense, and then I will merge

**Pull request merged**